pkg/cmd/roachtest/operations/backup_online_restore.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package operations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+type backupOnlineRestoreCleanup struct {
+	db string
+}
+
+func (cl *backupOnlineRestoreCleanup) Cleanup(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) {
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+
+	o.Status(fmt.Sprintf("dropping newly created db %s", cl.db))
+	_, err := conn.ExecContext(ctx, fmt.Sprintf("DROP DATABASE %s CASCADE", cl.db))
+	if err != nil {
+		o.Fatal(err)
+	}
+}
+
+func runbackupOnlineRestore(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) registry.OperationCleanup {
+	// This operation looks for the district table in a database named cct_tpcc or tpcc.
+	rng, _ := randutil.NewPseudoRand()
+	dbWhitelist := []string{"cct_tpcc", "tpcc"}
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+	dbs, err := conn.QueryContext(ctx, "SELECT database_name FROM [SHOW DATABASES]")
+	if err != nil {
+		o.Fatal(err)
+	}
+	var dbName string
+outer:
+	for dbs.Next() {
+		var dbStr string
+		if err := dbs.Scan(&dbStr); err != nil {
+			o.Fatal(err)
+		}
+		for i := range dbWhitelist {
+			if dbWhitelist[i] == dbStr {
+				// We found a db in the whitelist.
+				dbName = dbStr
+				break outer
+			}
+		}
+	}
+	if dbName == "" {
+		o.Status(fmt.Sprintf("did not find a db in the whitelist %v", dbWhitelist))
+		return nil
+	}
+
+	backupTS := timeutil.Now().Add(-1 * time.Minute).UTC().Format(time.DateTime)
+
+	o.Status(fmt.Sprintf("backing up table district in db %s", dbName))
+	bucket := fmt.Sprintf("gs://%s/operation-backup-restore/%d/?AUTH=implicit", testutils.BackupTestingBucket(), timeutil.Now().UnixNano())
+
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("BACKUP TABLE %s.district TO '%s' AS OF SYSTEM TIME '%s'", dbName, bucket, backupTS))
```

Comment:
could we just run the online restore after the regular restore above, so we avoid conducting two 100TB+ backups?

Reply:
we _could_, but I kind of want to leave the tests separate for now. The offline restore won't complete at all, for instance. And the extra 100TB is fine if it's only for a few days (which I assume it will be, but I'll check)